### ISSUE TITLE
Email throwing up an error as a keyword in snowflake

### ISSUE THREAD

**Issue by blesschianumba** [sqlfluff/sqlfluff#5235]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a field called 'email' that broke our company's SQLfluff linting pipeline. We got around it by using `--noqa: RF04`.

### Expected Behaviour

Email is not a keyword in Snowflake, and so should not throw an error.

### Observed Behaviour

When running `sqlfluff lint [model]`, I saw the following:

```
L:  19 | P:  88 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
```

### How to reproduce

Create a model with a field called email and lint it with `sqlfluff lint`

### Dialect

PostgreSQL

### Version

`sqlfluff, version 2.3.1`
`Python 3.11.4`

```
dbt Core:
  - installed: 1.6.0
  ```

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-finley** [comment]
Could you give an example query we could test it again? Thanks!

**Comment by rogalski** [comment]
Same goes for NAME.

Minimal repro:

```sql
SELECT 1 AS NAME;
```

Output:
```
(.venv) lrogalski@lrogalski-ThinkPad-E590:~/repo/bain/ng-python-HELIX-airflow-etl$ sqlfluff lint --dialect snowflake repro.sql 
== [repro.sql] FAIL                                                                                                                                                                                                          
L:   1 | P:  13 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
All Finished 📜 🎉!
```

Docs seems to explicitly document opinionated decision to disallow valid code:
https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/rules/references/RF04.py#L14C1-L17

>   Although `unreserved` keywords `can` be used as identifiers,
    and `reserved words` can be used as quoted identifiers,
    best practice is to avoid where possible, to avoid any
    misunderstandings as to what the alias represents.

IMHO either we need improved lint message (mention that keyword is unreserved) and possibly some feature flag to allow unreserved keywords as names and disallow only reserved keywords.

**Comment by WittierDinosaur** [comment]
These are unreserved keywords in the snowflake dialect. Closing as this is intended behaviour. The solution is to disable the rule if you don't want these lint errors.
